python-checks/src/test/resources/checks/mandatoryFunctionParameterTypeHint.py

```diff
@@ -0,0 +1,70 @@
+def foo(param): # Noncompliant {{Add a type hint to this function parameter.}}
+       #^^^^^
+    pass
+
+def foobar_multiline(
+        param1, # Noncompliant {{Add a type hint to this function parameter.}}
+       #^^^^^^
+        param2): # Noncompliant {{Add a type hint to this function parameter.}}
+       #^^^^^^
+    pass
+
+def foobar(param1, param2): # Noncompliant 2
+    pass
+
+class Bar:
+    def __init__(self, param): # Noncompliant {{Add a type hint to this function parameter.}}
+                      #^^^^^
+        pass
+
+    def foo(param): # Noncompliant {{Add a type hint to this function parameter.}}
+           #^^^^^
+        pass
+
+    def not_class_method(cls): # Noncompliant {{Add a type hint to this function parameter.}}
+                        #^^^
+        pass
+
+    @classmethod
+    def class_method(cls, param): # Noncompliant {{Add a type hint to this function parameter.}}
+                         #^^^^^
+        pass
+
+def nested(param1: str):
+    def foo(param2): # Noncompliant {{Add a type hint to this function parameter.}}
+           #^^^^^^
+        pass
+    print("the end")
+
+def dynamic_param_list(*args, **kwargs): # Noncompliant 2
```

Comment:
I think we should also add test cases where we have some "regular" parameters next to `*args` and `**kwargs`, and when they are not together.

```python
def foo(a, b, *args, **kwargs): ...
def bar(a, *args): ...
def baz (a, **kwargs): ...
```

Also  some parameters with a default value:
```python
def foo(x=None):
   pass
```